modules/core/src/test/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridCacheDhtPreloadWaitForBackupsTest.java

```diff
@@ -252,6 +273,235 @@ public void testShutdownWithoutBackups() throws Exception {
+            DistributedMetaStorageImpl.IGNITE_INTERNAL_KEY_PREFIX + "graceful.shutdown",
+            new HashSet<>(Collections.singleton(grid(0).localNode().id())));
+
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        UUID node1Id = grid(1).localNode().id();
+
+        Thread stopper = new Thread(() -> {
+            grid(1).close();
+            latch.countDown();
+        }, "Stopper");
+
+        stopper.start();
+
+        assertTrue(GridTestUtils.waitForCondition(() -> latch.getCount() == 0, STOP_TIMEOUT_LIMIT));
+
+        HashSet<UUID> expMetastoreContent = new HashSet<>();
+        expMetastoreContent.add(grid(0).localNode().id());
+        expMetastoreContent.add(node1Id);
+
+        assertEquals(
+            expMetastoreContent,
+            grid(0).context().distributedMetastorage().read(GRACEFUL_SHUTDOWN_METASTORE_KEY));
+    }
+
+    /**
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testThatExcludedNodeListWithinMetastoreCleanedUpAfterUpdatingFullMap() throws Exception {
+        cacheMode = CacheMode.PARTITIONED;
+        atomicityMode = CacheAtomicityMode.ATOMIC;
+        synchronizationMode = CacheWriteSynchronizationMode.FULL_ASYNC;
+        rebalanceMode = CacheRebalanceMode.ASYNC;
+        clientNodes = false;
+        backups = 3;
+
+        startGrids(4);
+
+        grid(0).cluster().active(true);
+
+        grid(0).context().distributedMetastorage().write(
+            DistributedMetaStorageImpl.IGNITE_INTERNAL_KEY_PREFIX + "graceful.shutdown",
+            new HashSet<>(Collections.singleton(grid(0).localNode().id())));
+
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        UUID node1Id = grid(1).localNode().id();
+
+        Thread stopper = new Thread(() -> {
+            grid(1).close();
+            latch.countDown();
+        }, "Stopper");
+
+        stopper.start();
+
+        assertTrue(GridTestUtils.waitForCondition(() -> latch.getCount() == 0, STOP_TIMEOUT_LIMIT));
+
+        HashSet<UUID> expMetastoreContent = new HashSet<>();
+        expMetastoreContent.add(grid(0).localNode().id());
+        expMetastoreContent.add(node1Id);
+
+        assertEquals(
+            expMetastoreContent,
+            grid(0).context().distributedMetastorage().read(GRACEFUL_SHUTDOWN_METASTORE_KEY));
+
+        UUID node2Id = grid(2).localNode().id();
+
+        grid(2).close();
+
+        expMetastoreContent.remove(node1Id);
+        expMetastoreContent.add(node2Id);
+
+        assertEquals(
+            expMetastoreContent,
+            grid(0).context().distributedMetastorage().read(GRACEFUL_SHUTDOWN_METASTORE_KEY));
+    }
+
+    /**
+     * @throws Exception If failed.
+     */
+    @Test
+    public void testClientNodeShouldStopImmediately() throws Exception {
+        cacheMode = CacheMode.PARTITIONED;
+        atomicityMode = CacheAtomicityMode.ATOMIC;
+        synchronizationMode = CacheWriteSynchronizationMode.PRIMARY_SYNC;
+        backups = 1;
```

Comment:
1 backup and 1 node is suspicious as well.

Reply:
1 backup and 1 server node, means that it should not be possible to stop server node with graceful shutdown.